src/code/z_demo.c

```diff
@@ -1232,683 +1401,778 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
                 play->envCtx.screenFillColor[1] = 0;
                 play->envCtx.screenFillColor[2] = 0;
-                if (cmd->base == 10) {
-                    play->envCtx.screenFillColor[3] = (1.0f - temp) * 255.0f;
+
+                if (cmd->type == CS_TRANS_BLACK_FILL) {
+                    play->envCtx.screenFillColor[3] = (1.0f - lerp) * 255.0f;
                 } else {
-                    play->envCtx.screenFillColor[3] = 255.0f * temp;
+                    play->envCtx.screenFillColor[3] = 255.0f * lerp;
                 }
                 break;
 
-            case 12:
-                gSaveContext.cutsceneTransitionControl = 255.0f - (155.0f * temp);
+            case CS_TRANS_BLACK_HALF_FILL:
+                gSaveContext.cutsceneTransitionControl = 255.0f - (155.0f * lerp);
                 break;
 
-            case 13:
+            case CS_TRANS_BLACK_HALF_UNFILL:
                 play->envCtx.screenFillColor[0] = 0;
                 play->envCtx.screenFillColor[1] = 0;
                 play->envCtx.screenFillColor[2] = 0;
-                play->envCtx.screenFillColor[3] = 255.0f - ((1.0f - temp) * 155.0f);
+                play->envCtx.screenFillColor[3] = 255.0f - ((1.0f - lerp) * 155.0f);
                 break;
         }
     }
 }
 
-// Command 0x1 & 0x5: Camera Eye Points
-s32 Cutscene_Command_CameraEyePoints(PlayState* play, CutsceneContext* csCtx, u8* cmd, u8 relativeToLink) {
-    s32 shouldContinue = 1;
-    CsCmdBase* cmdBase = (CsCmdBase*)cmd;
+s32 CutsceneCmd_SetCamEyePoints(PlayState* play, CutsceneContext* csCtx, u8* scriptPtr, u8 relativeToPlayer) {
+    s32 shouldContinue = true;
+    CsCmdGeneric* cmd = (CsCmdGeneric*)scriptPtr;
     s32 size;
 
-    cmd += 8;
-    size = 8;
+    scriptPtr += sizeof(CsCmdGeneric);
+    size = sizeof(CsCmdGeneric);
+
+    if ((csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame) &&
+        ((csCtx->camEyePointsAppliedFrame < cmd->startFrame) || (csCtx->camEyePointsAppliedFrame >= 0xF000))) {
+        csCtx->camEyeReady = true;
+        csCtx->camEyePoints = (CutsceneCameraPoint*)scriptPtr;
 
-    if ((cmdBase->startFrame < csCtx->frames) && (csCtx->frames < cmdBase->endFrame) &&
-        ((csCtx->unk_18 < cmdBase->startFrame) || (csCtx->unk_18 >= 0xF000))) {
-        csCtx->unk_1B = 1;
-        csCtx->subCamEyePoints = (CutsceneCameraPoint*)cmd;
-        if (csCtx->unk_1A != 0) {
-            csCtx->unk_18 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+        if (csCtx->camAtReady) {
+            // the frame number set here isn't important, it just signals that the camera data has been applied
+            csCtx->camEyePointsAppliedFrame = cmd->startFrame;
+
+            if (gUseCutsceneCam) {
                 Play_CameraChangeSetting(play, csCtx->subCamId, CAM_SET_CS_0);
-                Play_ChangeCameraStatus(play, sReturnToCamId, CAM_STAT_WAIT);
+                Play_ChangeCameraStatus(play, sPrevCamId, CAM_STAT_WAIT);
```

Comment:
Also from the subCam cleanup PR ( @engineer124 )
Choosing `sReturnToCamId` over `sPrevCamId` was a conscious choice but I forgot why...

... so I looked for the reason:
https://github.com/zeldaret/oot/pull/989#discussion_r776446117
> Would this->returnToCamId work in this case? This would apply to en_ta, en_kz, and en_in as they all have this same pattern and current name

and `returnToCamId` comes from a compromise:
https://github.com/zeldaret/oot/pull/989#discussion_r744261864
@Roman971